weather/src/js/api/weather-api.js

```diff
@@ -0,0 +1,80 @@
+import axios from 'axios';
+
+export function getWeatherData(units, coords) {
+    return new Promise(function(resolve, reject) {
+    
+    
+    axios.get('http://api.openweathermap.org/data/2.5/forecast',{
+        params : {
+                units: 'metric',
+                lat: coords.lat,
+                lon: coords.lon,
+                appid: '11ad38fb16a7a484238ba3e7396d2cf6'
+            }
+        })
+        .then(weatherData => {  
```

Comment:
There is too much logic happening within the `then` clause. Ideally you should decompose everything into smaller function that each does one thing